omod-1.10/src/main/java/org/openmrs/module/attachments/rest/AttachmentResource1_10.java

```diff
@@ -65,6 +77,39 @@ public Attachment getByUniqueId(String uniqueId) {
 		}
 	}
 	
+	/**
+	 * Gets attachements by patient or encounter (paged according to context if necessary) only if a
+	 * patient or encounter parameter exists respectively in the request set on the
+	 * {@link RequestContext} otherwise searches for obs that match the specified query
+	 *
+	 * @param context
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource#doSearch(org.openmrs.module.webservices.rest.web.RequestContext)
+	 */
+	@Override
+	protected PageableResult doSearch(RequestContext context) {
+		String patientUuid = context.getRequest().getParameter("patient");
+		if (patientUuid != null) {
```

Comment:
See what's done in the upload part, basically we need the patient UUID, if it is not provided we will have to throw an invalid request exception.

Reply:
Look at what's done in the upload routine of the REST resource.

Reply:
do I need to add if conditions which check the provided parameter is patient or encounter?

Reply:
Patient should always be there. You will need something a bit similar to [this](https://github.com/openmrs/openmrs-module-attachments/blob/1ed36e07e4a4aee0b908ef690058ad5053f3b371/omod-1.10/src/main/java/org/openmrs/module/attachments/rest/AttachmentResource1_10.java#L98-L100) if not `Patient` object could be obtained.